test/JDBC/input/datatypes/TestDatetime-vu-verify.txt

```diff
@@ -1 +1,35 @@
 SELECT * FROM TestDatetime_vu_prepare_dt;
+
+select convert(datetime, '2022-11-10 09:15:30.000');
```

Comment:
I assume we have validate all of these against SQL server, or better to generate file against SQL server.

Reply:
SQL query:
select convert(datetime, '2022-11-10 09:15:30.000');
select convert(datetime, '2022-11-10 09:15:30.001');
select convert(datetime, '2022-11-10 09:15:30.002');
select convert(datetime, '2022-11-10 09:15:30.003');
select convert(datetime, '2022-11-10 09:15:30.004');
select convert(datetime, '2022-11-10 09:15:30.005');
select convert(datetime, '2022-11-10 09:15:30.006');
select convert(datetime, '2022-11-10 09:15:30.007');
select convert(datetime, '2022-11-10 09:15:30.008');
select convert(datetime, '2022-11-10 09:15:30.009');
select convert(datetime, '2022-11-10 09:15:30.999');
select convert(datetime, '2022-11-10 21:26:45.120');
select convert(datetime, '2022-11-10 21:26:45.341');
select convert(datetime, '2022-11-10 21:26:45.562');
select convert(datetime, '2022-11-10 21:26:45.783');
select convert(datetime, '2022-11-10 21:26:45.904');
select convert(datetime, '2022-11-10 21:26:45.215');
select convert(datetime, '2022-11-10 21:26:45.436');
select convert(datetime, '2022-11-10 21:26:45.657');
select convert(datetime, '2022-11-10 21:26:45.878');
select convert(datetime, '2022-11-10 21:26:45.909');
select convert(datetime, '2000-12-13 12:58:23.123');
select convert(datetime, '2000-12-13 12:58:23.223');
select convert(datetime, '2000-12-13 12:58:23.323');
select convert(datetime, '2000-12-13 12:58:23.423');
select convert(datetime, '2000-12-13 12:58:23.523');
select convert(datetime, '2000-12-13 12:58:23.623');
select convert(datetime, '2000-12-13 12:58:23.723');
select convert(datetime, '2000-12-13 12:58:23.823');
select convert(datetime, '2000-12-13 12:58:23.923');
select convert(datetime, '2000-12-13 12:58:23.023');
select convert(datetime, '1753-01-01 23:59:59.997');
select convert(datetime, '1753-12-31 23:59:59.998');
select convert(datetime, '1753-12-31 23:59:59.999');
select convert(datetime, '9999-12-31 23:59:59.998');
select convert(datetime, '9999-12-31 23:59:59.999'); -- out of range
select convert(datetime, '1752-12-31 23:59:59.997');  -- out of range
select convert(datetime, '0000-00-00 00:00:00.000');  -- out of range

Datetime valid range: ('1753-01-01' To '9999-12-31')

SQL response:
2022-11-10 09:15:30.000
2022-11-10 09:15:30.000
2022-11-10 09:15:30.003
2022-11-10 09:15:30.003
2022-11-10 09:15:30.003
2022-11-10 09:15:30.007
2022-11-10 09:15:30.007
2022-11-10 09:15:30.007
2022-11-10 09:15:30.007
2022-11-10 09:15:30.010
2022-11-10 09:15:31.000
2022-11-10 21:26:45.120
2022-11-10 21:26:45.340
2022-11-10 21:26:45.563
2022-11-10 21:26:45.783
2022-11-10 21:26:45.903
2022-11-10 21:26:45.217
2022-11-10 21:26:45.437
2022-11-10 21:26:45.657
2022-11-10 21:26:45.877
2022-11-10 21:26:45.910
2000-12-13 12:58:23.123
2000-12-13 12:58:23.223
2000-12-13 12:58:23.323
2000-12-13 12:58:23.423
2000-12-13 12:58:23.523
2000-12-13 12:58:23.623
2000-12-13 12:58:23.723
2000-12-13 12:58:23.823
2000-12-13 12:58:23.923
2000-12-13 12:58:23.023
1753-01-01 23:59:59.997
1753-12-31 23:59:59.997
1754-01-01 00:00:00.000
9999-12-31 23:59:59.997
The conversion of a varchar data type to a datetime data type resulted in an out-of-range value.
The conversion of a varchar data type to a datetime data type resulted in an out-of-range value.
The conversion of a varchar data type to a datetime data type resulted in an out-of-range value.